nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/data/explorer/PutAzureDataExplorer.java

```diff
@@ -0,0 +1,309 @@
+        String mappingName = context.getProperty(MAPPING_NAME).getValue();
+        String routePartiallySuccessfulIngestion = context.getProperty(ROUTE_PARTIALLY_SUCCESSFUL_INGESTION).getValue();
+        String ingestionStatusPollingTimeout = context.getProperty(INGESTION_STATUS_POLLING_TIMEOUT).getValue();
+        String ingestionStatusPollingInterval = context.getProperty(INGESTION_STATUS_POLLING_INTERVAL).getValue();
+        String ignoreFirstRecord = context.getProperty(IGNORE_FIRST_RECORD).getValue();
+        streamingEnabled = context.getProperty(IS_STREAMING_ENABLED).evaluateAttributeExpressions().asBoolean();
+        pollOnIngestionStatus = context.getProperty(POLL_ON_INGESTION_STATUS).evaluateAttributeExpressions().asBoolean();
+
+        boolean isError = false;
+
+        // ingestion into ADX
+        try (final InputStream inputStream = session.read(flowFile)) {
+            StringBuilder ingestLogString = new StringBuilder().append("Ingesting with: ")
+                    .append("dataFormat - ").append(dataFormat).append("|")
+                    .append("ingestionMappingName - ").append(mappingName).append("|")
+                    .append("databaseName - ").append(databaseName).append("|")
+                    .append("tableName - ").append(tableName).append("|")
+                    .append("pollOnIngestionStatus - ").append(pollOnIngestionStatus).append("|")
+                    .append("ingestionStatusPollingTimeout - ").append(ingestionStatusPollingTimeout).append("|")
+                    .append("ingestionStatusPollingInterval - ").append(ingestionStatusPollingInterval).append("|")
+                    .append("routePartiallySuccessfulIngestion - ").append(routePartiallySuccessfulIngestion);
+            getLogger().debug(ingestLogString.toString());
+
+            KustoIngestionResult result = service.ingestData(
+                    new KustoIngestionRequest(streamingEnabled, pollOnIngestionStatus, inputStream, databaseName,
+                            tableName, dataFormat, mappingName, ignoreFirstRecord, ingestionStatusPollingTimeout, ingestionStatusPollingInterval)
+            );
+            getLogger().info("Ingest Status Polling Enabled {} and Ingest Status Polling Enabled {} ", pollOnIngestionStatus, result.toString());
+
+            if (result == KustoIngestionResult.SUCCEEDED) {
+                getLogger().info("Ingest Completed - {}", result.getStatus());
+            } else if (result == KustoIngestionResult.FAILED) {
+                getLogger().error("Ingest Failed - {}", result.getStatus());
+                isError = true;
+            } else if (result == KustoIngestionResult.PARTIALLY_SUCCEEDED) {
+                getLogger().warn("Ingest Partially succeeded - {}", result.getStatus());
+                flowFile = session.putAttribute(flowFile, "ingestion_status", "partial_success");
+                if (StringUtils.equalsIgnoreCase(routePartiallySuccessfulIngestion, "Failure")) {
```

Comment:
This should be compared using the failure relationship name instead of repeating the `Failure` string for the comparison.